src/dotty/tools/backend/jvm/DottyBackendInterface.scala

```diff
@@ -569,7 +569,7 @@ class DottyBackendInterface(outputDirectory: AbstractFile)(implicit ctx: Context
     def isConstructor: Boolean = toDenot(sym).isConstructor
     def isAnonymousFunction: Boolean = toDenot(sym).isAnonymousFunction
     def isMethod: Boolean = sym is Flags.Method
-    def isPublic: Boolean =  sym.flags.is(Flags.EmptyFlags, Flags.Private | Flags.Protected)
+    def isPublic: Boolean =  toDenot(sym).isPublic
```

Comment:
The common bakend does not.
I'm not sure why this is needed in scalac. I'm not in favor of such change in Dotty backend.
Nor do I think that we should add this incorrect implementation of `isPrivate` method: If method is defined as `private [package] def foo = ???` it has to be compiled as public. 
If some specific phase has different notion of `isPublic` - such implementation can go into this phase.